packages/colorpickers/src/elements/ColorPicker/index.tsx

```diff
@@ -214,59 +206,11 @@ export const ColorPicker = forwardRef<HTMLDivElement, IColorPickerProps>(
             <StyledInput
               isCompact
               maxLength={7}
-              value={hexInput}
+              value={state.hexInput}
               /* eslint-disable jsx-a11y/no-autofocus */
               autoFocus={autofocus}
-              onChange={e => {
-                setHexInput(e.target.value);
-                if (isValidHex(e.target.value)) {
-                  const rgb = parseToRgb(e.target.value);
-                  const hsl = rgbToHsl(rgb.red, rgb.green, rgb.blue);
-
-                  if (isControlled) {
-                    onChange &&
-                      onChange({
-                        hue: hsl.h,
-                        saturation: hsl.s,
-                        lightness: hsl.l,
-                        hex: e.target.value,
-                        red: rgb.red,
-                        green: rgb.green,
-                        blue: rgb.blue,
-                        alpha: state.alpha
-                      });
-                  } else {
-                    dispatch({ type: 'HEX_CHANGE', payload: e.target.value });
-                  }
-                }
-              }}
-              onBlur={e => {
-                if (!e.target.value.includes('#')) {
-                  const hexInputString = `#${e.target.value}`;
-
-                  if (isValidHex(hexInputString)) {
-                    setHexInput(hexInputString);
-                    const rgb = parseToRgb(hexInputString);
-                    const hsl = rgbToHsl(rgb.red, rgb.green, rgb.blue);
-
-                    if (isControlled) {
-                      onChange &&
-                        onChange({
-                          hue: hsl.h,
-                          saturation: hsl.s,
-                          lightness: hsl.l,
-                          hex: hexInputString,
-                          red: rgb.red,
-                          green: rgb.green,
-                          blue: rgb.blue,
-                          alpha: state.alpha
-                        });
-                    } else {
-                      dispatch({ type: 'HEX_CHANGE', payload: hexInputString });
-                    }
-                  }
-                }
-              }}
+              onBlur={handleBlur}
```

Comment:
We now reset values based on the computed color whenever a blur event occurs. This cleans up _valid_ number inputs like `e` and other non-int notations.